lib/Client.js

```diff
@@ -149,6 +182,23 @@ class Client {
         this.client.editStatus("online", { name });
     }
 
+    clientReporting() {
+        this.database.insertStat("messageRate", this.state.messageCount);
+        this.database.insertStat("userCount", this.client.users.size);
+        this.database.insertStat("memoryUsage", process.memoryUsage().heapUsed);
+        this.database.insertStat("databaseLatency", Math.round(_.mean(this.database.latestLatencies)));
+        this.database.insertStat("shardLatency", this.getAverageShardLatency());
```

Comment:
[Optional] I don't know if we'd ever need those stat type names elsewhere, but if we did, it would be great if those were in constants or something. Not really necessary for now since we don't, but might be worth actively adding them as constants instead of moving them the day we do need them.

Reply:
The way I see it, we should probably stick to one source of data for each stat type. If it's going elsewhere, then it will probably end up getting removed from here.